### PR TITLE
Added support for ARM based Open ESC firmware.

### PR DESCRIPTION
**Pull request by mikeller**
Adds support for flashing open source Open ESC firmware from https://github.com/AlkaMotors/AM32-MultiRotor-ESC-firmware onto ARM F051 based ESCs.

### CODE REVIEW

File: js/open_esc_escs.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    "layouts": {
+        "Arm": {
+            "FD6288": { "name": "FD6288 based", "fileName": "FD6288" },
+            "HGLRC60A": { "name": "HGLRC 60A", "fileName": "HGLRC60A" },
+            "IFLIGHT": { "name": "iFlight 50A", "fileName": "IFLIGHT" },
+            "MP6531": { "name": "MP6531 based", "fileName": "MP6531" },
+            "SISKIN": { "name": "Sunrise Siskin", "fileName": "SISKIN" },
+            "TMOTOR45A": { "name": "T-motor 45A", "fileName": "TMOTOR45A" },
+            "TMOTOR55A": { "name": "T-motor 55A", "fileName": "TMOTOR55A" }
```
Comment (DusKing1) [inline js/open_esc_escs.json:10]: So if later I add more targets, this needs to change to match up as well?

**Review by DusKing1 — COMMENTED** [review]
Just a question...

File: js/open_esc_escs.json (hunk 1 above)
Comment (mikeller) [reply js/open_esc_escs.json:10]: Yes. But this file is downloaded from GitHub every time configurator is run, so if it is updated users will get the changes without having to download and install an updated version of configurator.

**Comment by DusKing1** [comment]
I found a bug and I wanna report it.

### The process bar stops at 80% and reports flash failed. But the firmware was flash successfully. This "failed" message is really misleading.

How to reproduce:

- Hardware: iflight general ESC target
- Base firmware: custom build, version 1.60
- Target firmware: custom build, version 1.70 (just a random pick, no meaning)

1. Connect to bl-configurator, read all
2. Click "Flash all"
3. Select 1.70 firmware hex, then flash

When the process bar comes to about 80%, the bar will stop and the configurator will show a failed message. Like this:
image

When flashing finished, the ESC's version in the configurator will still be shown as 1.60.

If we click "Read setup" at this time, the ESC will become "NOT READY".
image

Give it a disconnected and reconnected, then the 1.70 will show up.
image

This flash failed message is really misleading. I don't know if this is a bug, but if not better make some note about the flash procedure, otherwise we might receive lots of flashing failed reports.

**Comment by mikeller** [comment]
@DusKing1: Yes, I get this too. I will have to investigate, this did not happen during development. But anyhow, the error just happens at the end of the verification, so this can safely be used for testing.

**Comment by mikeller** [comment]
Merging this now so further bugfixing can be done with updated libraries and `yarn`.

**Pull request merged**